example/jitter/memstruct.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python
+"""This script is just a short example of common usages for miasm2.analysis.mem.
+For a more complete view of what is possible, tests/analysis/mem.py covers
+most of the module possibilities, and the module doc gives useful information
+as well.
+"""
+
+from miasm2.analysis.machine import Machine
+from miasm2.analysis.mem import MemStruct, MemSelf, MemVoid, MemStr,\
+                                MemSizedArray, Ptr, Num, Array, set_allocator
+from miasm2.os_dep.common import heap
+
+# Instanciate a heap
+my_heap = heap()
+# And set it as the default memory allocator, to avoid manual allocation and
+# explicit address passing to the MemStruct constructor
+set_allocator(my_heap.vm_alloc)
+
+# Let's reimplement a simple C generic linked list mapped on a VmMngr!
+
+# All the structures and methods will use the python objects but all the data
+# is in fact stored in the VmMngr
+
+class ListNode(MemStruct):
+    fields = [
+        # The "<I" is the struct-like format of the pointer in memory, in this
+        # case a Little Endian 32 bits unsigned int
+        # One way to handle reference to ListNode in ListNode is to use the
+        # special marker MemSelf.
+        # You could also set or modify ListNode.fields after the class
+        # declaration and call ListNode.gen_fields()
+        ("next", Ptr("<I", MemSelf)),
+        # Ptr(_, MemVoid) is analogous to void*, MemVoid is just an empty
+        # MemStruct type
+        ("data", Ptr("<I", MemVoid)),
+    ]
+
+    def get_next(self):
+        if self.next == 0:
+            return None
+        return self.deref_next
+
+    def get_data(self, data_type=None):
+        if data_type is not None:
+            return self.deref_data.cast(data_type)
+        else:
+            return self.deref_data
+
+
+class LinkedList(MemStruct):
+    fields = [
+        ("head", Ptr("<I", ListNode)),
+        ("tail", Ptr("<I", ListNode)),
+        # Num can take any one-field struct-like format, including floats and
+        # doubles
+        ("size", Num("<I")),
+    ]
+
+    def __init__(self, vm, *args, **kwargs):
```

Comment:
Please choose between the form `(*args, **kwargs)` and `(vm, addr, *args, **kwargs)`